sdk/client-core/http-jdk-httpclient/src/main/java/io/clientcore/http/jdk/httpclient/implementation/BodyPublisherUtils.java

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.http.jdk.httpclient.implementation;
+
+import io.clientcore.core.http.models.HttpHeaderName;
+import io.clientcore.core.util.binarydata.BinaryData;
+import io.clientcore.core.util.binarydata.ByteArrayBinaryData;
+import io.clientcore.core.util.binarydata.SerializableBinaryData;
+import io.clientcore.core.util.binarydata.StringBinaryData;
+
+import java.net.http.HttpRequest;
+import java.time.Duration;
+
+import static java.net.http.HttpRequest.BodyPublishers.fromPublisher;
+import static java.net.http.HttpRequest.BodyPublishers.noBody;
+
+/**
+ * Utility class for BodyPublisher.
+ */
+public final class BodyPublisherUtils {
+    private BodyPublisherUtils() {
+    }
+
+    /**
+     * Creates BodyPublisher depending on underlying body content type.
+     * If progress reporter is not null, configures it to track request body upload.
+     *
+     * @param request {@link io.clientcore.core.http.models.HttpRequest} instance
+     * @param writeTimeout write timeout
+     * @return the request BodyPublisher
+     */
+    public static HttpRequest.BodyPublisher toBodyPublisher(io.clientcore.core.http.models.HttpRequest request,
+        Duration writeTimeout) {
+        // TODO (alzimmer): azure-core was using Flux.timeout to handle write timeouts. The logic will need to be
+        //  re-implemented to handle write timeouts in a similar manner.
```

Comment:
Do we want to have `writeTimeout` if it is not supported by the underlying HTTP client?

Reply:
I think this can be added as a follow-up